mava/wrappers/robot_warehouse.py

```diff
@@ -0,0 +1,227 @@
+        truncated = np.array(truncated).all()
+        self._done = terminated or truncated
+        if self._done:
+            self._reset_next_step = True
+
+        # Get legal actions
+        actions_mask = self._get_actions_mask()
+
+        # Convert observations and legal actions to numpy arrays
+        next_observations, actions_mask = self._convert_observations(
+            next_observations, actions_mask
+        )
+
+        # Reward info
+        if self.team_reward:
+            reward = np.array([np.array(reward).mean()] * self.num_agents)
+        else:
+            reward = np.array(reward)
+
+        # State info
+        info = {"extra_info": self._info, "actions_mask": actions_mask}
+
+        return next_observations, reward, terminated, truncated, info
+
+    def _get_actions_mask(self) -> List:
+        """Get legal actions from the environment."""
+        actions_mask = []
+        for agent_id in range(self.n_agents):
+            avail_agent = self.get_avail_agent_actions(agent_id)
+            actions_mask.append(avail_agent)
+        return actions_mask
+
+    def get_avail_agent_actions(self, agent_id: int) -> List:
+        """Returns the available actions for agent_id"""
+        # NOTE: robot-warehouse has no concept of legal actions
+        return [1] * self.env.action_space[agent_id].n
+
+    def _convert_observations(
+        self, observations: List, actions_mask: List
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Convert observation to it's numpy array."""
+        converted_observations: np.ndarray = np.array(observations, dtype="float32")
+        converted_actions_mask: np.ndarray = np.array(actions_mask, dtype="float32")
+
+        return converted_observations, converted_actions_mask
+
+
+class AgentIDWrapper(gym.Wrapper):
+    """Add onehot agent IDs to observation."""
+
+    def __init__(self, env: gym.Env):
+        super().__init__(env)
+
+        self.agent_ids = np.eye(self.env.num_agents)
+        _obs_low, _obs_high, _obs_dtype, _obs_shape = (
+            self.env.observation_space.low[0][0],
+            self.env.observation_space.high[0][0],
+            self.env.observation_space.dtype,
+            self.env.observation_space.shape,
+        )
+        _new_obs_shape = (self.env.num_agents, _obs_shape[1] + self.env.num_agents)
+
+        self.observation_space = Box(
+            low=_obs_low, high=_obs_high, shape=_new_obs_shape, dtype=_obs_dtype
+        )
+
+    def reset(self) -> Tuple[np.ndarray, Dict]:
+        """Reset the environment."""
+        obs, info = self.env.reset()
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, info
+
+    def step(self, action: list) -> Tuple[np.ndarray, float, bool, bool, Dict]:
+        """Step the environment."""
+        obs, reward, terminated, truncated, info = self.env.step(action)
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, reward, terminated, truncated, info
+
+
```

Comment:
Couple other comments here, it would be nice if we could integrate this with @WiemKhlifi new make function.

Do we need to set 
```python
       envs.num_envs = num_envs
       envs.is_vector_env = True
```
Do we use them? I think `num_envs` is set automatically anyways